apps/parsley/src/hooks/useOpenSectionAndScrollToLine/index.ts

```diff
@@ -0,0 +1,64 @@
+import { useEffect, useState } from "react";
+import { UseSectionsResult } from "hooks/useSections";
+import { ProcessedLogLines } from "types/logs";
+import { findLineIndex } from "utils/findLineIndex";
+
+/**
+ * useOpenSectionAndScrollToLine hook is used to open sections containing given line number(s) and scroll to the first given line number.
+ * This hook is designed to scroll even if sections are disabled.
+ * @param processedLogLines These are the visible log lines that are currently rendered in the log pane.
+ * @param openSectionContainingLineNumber This function is used to open the section containing the line number. The function returns true if
+ * the section is already open and false otherwise.
+ * @param scrollToLine This function is used to scroll to the line number.
+ * @returns A setter function that accepts one or many line numbers so scroll to. If multiple lines numbers are passed, open
+ * their corresponding sections and scroll to the first line.
+ */
+const useOpenSectionAndScrollToLine = (
+  processedLogLines: ProcessedLogLines,
+  openSectionContainingLineNumber: UseSectionsResult["openSectionContainingLineNumber"],
+  scrollToLine: (lineNumber: number) => void,
+) => {
+  const [lineNumber, setLineNumber] = useState<number | number[] | undefined>();
```

Comment:
Could we rename this variable? It's quite confusing that it can be singular or plural. `lineNumber` implies that it would be a single number

Reply:
Based on the conversation above, is there a reason why it can't always just be `number[]`?

Reply:
This can be set to array. I think it's convenient that this function accepts a single number or number[] to avoid having to cast to an array in other parts of the application. I pushed up the array casting into this hook as separate setter function. Let me know what you think about this commit: https://github.com/evergreen-ci/ui/pull/268/commits/67f70f89ed798e315b6dc42549e292f4c0f80681